File: ZhigaoWeather/app/src/java/com/weather/zhigao/adapter/divider/SpacesItemDecoration.java
package com.weather.zhigao.adapter.divider;

import android.content.Context;
import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.weather.zhigao.utils.ScreenUtils;


public class SpacesItemDecoration extends RecyclerView.ItemDecoration {

    private int space;

    public SpacesItemDecoration(Context context, int space) {
        this.space = ScreenUtils.dp2px(context,space);
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.left = space;
        outRect.right = space;
        outRect.bottom = space;

        // Add top margin only for the first item to avoid double space between items
        if (parent.getChildLayoutPosition(view) == 0) {
            outRect.top = space;
        } else {
            outRect.top = 0;
        }
    }
}

File: ZhigaoWeather/app/src/main/java/com/weather/zhigao/adapter/divider/SpacesItemDecoration.kt
package com.weather.zhigao.adapter.divider

import android.content.Context
import android.graphics.Rect
import android.support.v7.widget.RecyclerView
import android.view.View

import com.weather.zhigao.utils.ScreenUtils


class SpacesItemDecoration(context: Context, space: Int) : RecyclerView.ItemDecoration() {

    private val space: Int

    init {
        this.space = ScreenUtils.dp2px(context, space.toFloat())
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        outRect.left = space
        outRect.right = space
        outRect.bottom = space

        // Add top margin only for the first item to avoid double space between items
        if (parent.getChildLayoutPosition(view) == 0) {
            outRect.top = space
        } else {
            outRect.top = 0
        }
    }
}